bundles/org.eclipse.jface/src/org/eclipse/jface/operation/ModalContext.java

```diff
@@ -435,7 +435,9 @@ private static void runInCurrentThread(IRunnableWithProgress runnable, IProgress
 				runnable.run(progressMonitor);
 			}
 		} catch (OperationCanceledException e) {
-			throw new InterruptedException();
+			InterruptedException interruptedException = new InterruptedException(e.getLocalizedMessage());
```

Comment:
I miss a systematic approach here. In the same file `org.eclipse.jface.operation.ModalContext.run(IRunnableWithProgress, boolean, IProgressMonitor, Display)` also created an InterruptedException. Why isn't that code path updated?